src/libYARP_dev/include/yarp/dev/IFrameSource.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_DEV_IFRAMESOURCE_H
+#define YARP_DEV_IFRAMESOURCE_H
+
+#include <unordered_set>
+#include <map>
+#include <vector>
+#include <functional>
+#include <robotology/Result.h>
+#include <yarp/dev/api.h>
+#include <yarp/math/Quaternion.h>
+#include <yarp/math/FrameTransform.h>
+
+namespace yarp {
+    namespace dev {
+        class IFrameSource;
+    }
+}
+
+struct CustomFrameCompare
+{
+    bool operator()(const yarp::math::FrameTransform& lhs, const yarp::math::FrameTransform& rhs)
+    {
+        if (lhs.frameId == rhs.frameId)
+            return true;
+    }
+};
+
+class FrameEditor
+{
+    std::map<std::string, yarp::math::FrameTransform> storage;
+    yarp::math::FrameTransform nullframe;
+public:
+    const auto begin() const { return storage.begin(); }
```

Comment:
At least in the  definition of a class in the public API, I think it is better to avoid using `auto`, otherwise the programmer has no way of understanding the type returned by the method.

Reply:
and it is right. the return value of the method is strictly dependent on the return value of `std::map::begin()` and should remain so. it is also most likely that the return value will not be directly used by the user but passed to std method or called automatically by the compiler in the range-based for loops.


examples are `std::transform(myframeeditor.begin(), … etc)` or `for (const auto& f : myframeeditor){}`

Reply:
what the programmer have to understand is that begin method is a propagation of std::map::begin method. no other information are needed